src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -181,3 +187,56 @@ def _try_write_trace_to_cosmosdb(
         stack_trace = traceback.format_exc()
         logger.error(f"Failed to write trace to cosmosdb: {e}, stack trace is {stack_trace}")
         return
+
+
+def _try_get_blob_container_client(
+    logger: logging.Logger,
+    subscription_id: str,
+    resource_group_name: str,
+    workspace_name: str,
+    credential: Optional[object] = None,
+):
+    try:
+        from azure.ai.ml import MLClient
+        from azure.ai.ml._azure_environments import _get_storage_endpoint_from_metadata
+        from azure.ai.ml._restclient.v2022_10_01.models import DatastoreType
+        from azure.ai.ml.constants._common import STORAGE_ACCOUNT_URLS
+        from azure.storage.blob import ContainerClient
+
+        # To write data to blob, user should have "Storage Blob Data Contributor" to the storage account.
+        if credential is None:
+            from azure.identity import DefaultAzureCredential
+
+            credential = DefaultAzureCredential()
+
+        ml_client = MLClient(
+            credential=credential,
+            subscription_id=subscription_id,
+            resource_group_name=resource_group_name,
+            workspace_name=workspace_name,
+        )
+
+        default_datastore = ml_client.datastores.get_default()
+        if default_datastore.type != DatastoreType.AZURE_BLOB:
+            logger.error(f"Default datastore is {default_datastore.type}, not AzureBlob, skip getting blob client.")
+            return
+
+        storage_endpoint = _get_storage_endpoint_from_metadata()
+        account_url = STORAGE_ACCOUNT_URLS[DatastoreType.AZURE_BLOB].format(
+            default_datastore.account_name, storage_endpoint
+        )
+
+        from azure.ai.ml.constants._common import LONG_URI_FORMAT
+
+        # Datastore is a notion of AzureML, it is not a notion of Blob Storage.
+        # So, we cannot get datastore name by blob client.
+        # To generate the azureml uri has datastore name, we need to generate the uri here and pass in to db client.
+
+        return ContainerClient(
+            account_url=account_url, container_name=default_datastore.container_name, credential=credential
+        ), LONG_URI_FORMAT.format(subscription_id, resource_group_name, workspace_name, default_datastore.name, "")
+
+    except Exception as e:
+        stack_trace = traceback.format_exc()
```

Comment:
If fail to get the client, should we fail the request directly?

Reply:
Fixed. Will raise the exception.